File: python/hands_on_rl/bandit.py
import numpy as np


class Bandit:
    """Multi-armed bandit with Bernoulli rewards."""

    def __init__(self, n: int):
        self.n = n
        self.qs = np.random.uniform(size=n)
        self.best_q = np.max(self.qs)

    def __call__(self, action: int) -> int:
        return 1 if np.random.random() < self.qs[action] else 0

File: lib/bandit.dart
import 'random.dart' as r;
import 'dart:math';

class Bandit {
  final int n;
  final List<double> qs;
  late final double bestQ;

  Bandit(this.n) : qs = List.generate(n, (_) => r.random.nextDouble()) {
    bestQ = qs.reduce(max);
  }

  int call(int a) => r.random.nextDouble() < qs[a] ? 1 : 0;
}
